python/nav/portadmin/snmp/cisco.py

```diff
@@ -300,5 +315,88 @@ def get_dot1x_enabled_interfaces(self):
             for oid, state in self._bulkwalk(self.dot1xPortAuth)
         }
 
+    def get_poe_state_options(self) -> Sequence[PoeState]:
+        """Returns the available options for enabling/disabling PoE on this netbox"""
+        options_list = [
+            self.POE_AUTO,
+            self.POE_STATIC,
+            self.POE_LIMIT,
+            self.POE_DISABLE,
+        ]
+        return options_list
+
+    @translate_protocol_errors
+    def set_poe_state(self, interface: manage.Interface, state: PoeState):
+        """Set state for enabling/disabling PoE on this interface.
+        Available options should be retrieved using `get_poe_state_options`
+        """
+        unit_number, interface_number = self._get_poe_indexes_for_interface(interface)
+        oid_with_unit_number = self.POEENABLE + f".{unit_number}"
```

Comment:
```suggestion
        oid_with_unit_number = self.POEENABLE + OID((unit_number,))
```

Though, this reminds me that I was thinking the `OID` constructor should accept a single integer as valid input... i.e. I think the following should have been possible, but isn't right now:

```pycon
>>> OID(1)
OID('.1')
>>>
```